pkg/ansible/runner.go

```diff
@@ -0,0 +1,201 @@
+package ansible
+
+import (
+	"path"
+
+	log "github.com/sirupsen/logrus"
+
+	kbatch "k8s.io/api/batch/v1"
+	kapi "k8s.io/api/core/v1"
+	kapierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	openshiftAnsibleImage          = "openshift/origin-ansible:v3.7"
+	openshiftAnsibleServiceAccount = "cluster-operator-controller-manager"
+	inventoryConfigMap             = "ansible-inventory"
+	sshPrivateKeySecret            = "ssh-private-key"
+	operatorNamespace              = "cluster-operator"
```

Comment:
We can't hard-code this namespace. The user needs to be able to specify their own namespace in which to run cluster-operator.

Reply:
Now that I think about it more, I am leaning towards putting the.configmap and job into the namespace of the cluster anyway instead of the namespace of the controller manager. Are there reasons, such as ease of rbac, that make it better or easier if they are in the namespace of the controller manager?

Reply:
Agreed we'll need some way to configure this. Any thoughts on how? My last controller with online team I was using a configmap and mounting in as if it were a config file in the pod.

Are you envisioning that every cluster the CO manages gets it's own namespace? I was envisioning the opposite, that all data for all manages CO clusters lives in the cluster-operator namespace. I don't think we've ever actually discussed one way or the other.

One namespace per managed cluster has some implications for deployment, it would largely assume the cluster we run in is used only for cluster-operator I think, otherwise we might have some issues with namespace clutter and collisions. We might be able to swing that for our purposes but it could cause problems if we ever get to on-premise customer use cases. (a topic we all need to talk a lot more about to understand anyhow)

I'm not real familiar with pros/cons of either approach, will need to think on this and maybe we can discuss with team next chance we get.

Reply:
If the ansible container is required to run as root, one important consideration is that wherever it runs, it needs to have access to a service account that has the runAsAny SCC. Right now that is only the cluster-operator namespace. 

In my opinion, if this is something that will be run by the controller it's acceptable to run it in a special namespace where you have the service account you need. On the other hand, from an accounting/resource usage perspective, it'd be nice to track the resource usage of this job with the rest of the cluster resources for the user (in the namespace where those cluster resources live). Maybe it's worth investigating whether we really do need to run the ansible container as root instead of a random user.

Reply:
Any thoughts @abutcher ? I tried and got into a problem with ansible erroring out trying to look up the user identity for an obscure UID which looked like it was the fsGroupID for the namespace.

Reply:
@dgoodwin in the scl containers that require a valid username, we use libnss_wrapper to trick the main process into thinking that the users exists in passwd:
https://github.com/sclorg/postgresql-container/blob/master/9.5/root/usr/share/container-scripts/postgresql/common.sh#L137-L145

Reply:
I think we should try to run as non-root if we can. I'm digging into the current image now to see what it's up to.